core/dvector.h

```diff
@@ -232,17 +232,19 @@ class DVector {
 	}
 
 
-	void remove(int p_index) {
+	T remove(int p_index) {
 
+		T ret = get(p_index);
 		int s = size();
-		ERR_FAIL_INDEX(p_index, s);
+		ERR_FAIL_INDEX_V(p_index, s, T());
```

Comment:
This is where it fails (I think) when trying to return a value which is outside of bounds.